### PR TITLE
Fix InlineUIContainer clipping in text when ArrangeOverride called. (#17183)

### LINKED ISSUE #17183: `InlineUIContainer` dosen't correctly render with `TextProperty=Wrap`
### Describe the bug

Right now if Inlines have InlineUIContainer inside and we have TextProperty=Wrap on TextBlock, everything is fine until InlineUIContainer renders.
image
image

After that, if i back to smaller width, InlineUiContainer will be rendered incorrectly, on top of my text.
image

I don't know what to do right now. I thought about using WrapPanel, but that means i need to rewrote whole system in my App.

### To Reproduce

1. Create inline.
2. Add runs and InlineUIContainer.
3. Set inlines on some TextBlock with changable width.
4. Try to change size on TextBlock
5. InlineUIContainer will show and never hide.

### Expected behavior

1. Create inline.
2. Add runs and InlineUIContainer.
3. Set inlines on some TextBlock with changable width.
4. Try to change size on TextBlock
5. InlineUIContainer will proper show and hide as width changes.

### Avalonia version

11.1.3

### OS

Windows

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Danich27rus**
## What does the pull request do?

Issue #17183
This pr fixes problem with InlineUIContainer clipping inside a text and stuck on coordinates.

## What is the current behavior?

Currently InlineUIContainer will be rendered on top of other Run when Width of control will be changed.

## What is the updated/expected behavior with this PR?

InlineUIContainer will be rendered outside of textblock when it's not present in collection of inlines.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

## Breaking changes
Changes the behavior of Avalonia.ArrangeOverride int TextBlock.

## Fixed issues
Fixes #17183

### CODE REVIEW

**Comment by Danich27rus** [comment]
@cla-avalonia agree

**Comment by Danich27rus** [comment]
> @Danich27rus,
> 
> Please read the following Contributor License Agreement (CLA). If you agree with the CLA, please reply with the following:
> 
> ```
> @cla-avalonia agree
> ```
> 
> Contributor License Agreement

@cla-avalonia agree

File: src/Avalonia.Controls/TextBlock.cs [hunk 1]
```diff
@@ -727,6 +727,8 @@ protected override Size MeasureOverride(Size availableSize)
             return new Size(width, TextLayout.Height).Inflate(padding);
         }
 
+        List<Control> _embeddedControlRunControls = new();
```
Comment (Gillibald) [inline src/Avalonia.Controls/TextBlock.cs:730]: This isn't needed. VisualChildren holds all controls already.

**Comment by Gillibald** [comment]
Why do we need multiple arrange calls for embedded controls? There is some invalidation issue and we might need to invalidate visual children somewhere. I can't accept this "hack".

**Comment by Danich27rus** [comment]
> multiple arrange calls for embedded controls

It's a dirty hack, i understand that. I just don't understand how we can validate draw operation for inlines, if InlineUIContainer is not present in list of incoming runs.

**Comment by Gillibald** [comment]
Embedded controls are added as a visual child to TextBlock https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Documents/InlineCollection.cs#L185

**Comment by Danich27rus** [comment]
So i made a little research. `EmbeddedControlRun` is not present in `TextLayout` if it too big for `TextLine`, so basically if it's not present in `TextLines` - it cannot be changed in any way, only through VisualChidren. I don't think it's possible to validate child without changing `TextFormatterImpl.cs` internals

**Comment by Gillibald** [comment]
> So i made a little research. `EmbeddedControlRun` is not present in `TextLayout` if it too big for `TextLine`, so basically if it's not present in `TextLines` - it cannot be changed in any way, only through VisualChidren. I don't think it's possible to validate child without changing `TextFormatterImpl.cs` internals

Simply clear visual children in the arrange call and add them again before arranging the embedded runs.

**Comment by Danich27rus** [comment]
I (at least tried) to makle this like you described, but i'm not still sure that 100% correct solution

File: samples/ControlCatalog/Pages/TextBlockPage.xaml [hunk 2]
```diff
@@ -133,7 +133,7 @@
           <Run Text="ABC" FontFeatures="+c2sc, +smcp"/>
           <Run Text="DEF"/>
           <Run Text="0123" FontFeatures="frac"/>
-        </TextBlock> 
+        </TextBlock>
```
Comment (Gillibald) [inline samples/ControlCatalog/Pages/TextBlockPage.xaml:136]: Please revert

File: src/Avalonia.Controls/TextBlock.cs [hunk 3]
```diff
@@ -755,6 +758,17 @@ protected override Size ArrangeOverride(Size finalSize)
                             if (drawable is EmbeddedControlRun controlRun
                                 && controlRun.Control is Control control)
                             {
+                                //Add again to prevent clipping
+                                //Fixes: #17194
+                                foreach (var inline in Inlines!)
```
Comment (Gillibald) [inline src/Avalonia.Controls/TextBlock.cs:763]: Remove this entirely and instead call `VisualChildren.Add(control)`

File: samples/ControlCatalog/Pages/TextBlockPage.xaml (hunk 2 above)
Comment (Danich27rus) [reply samples/ControlCatalog/Pages/TextBlockPage.xaml]: Reverted

File: src/Avalonia.Controls/TextBlock.cs (hunk 3 above)
Comment (Danich27rus) [reply src/Avalonia.Controls/TextBlock.cs]: Call only `VisualChildren.Add(control)`

**Comment by Danich27rus** [comment]
@Gillibald thank you very mush for hekping. I'm sorry about my stupidity

**Review by Gillibald — APPROVED** [review]

**Review by Gillibald — APPROVED** [review]

**Comment by Danich27rus** [comment]
@cla-avalonia agree

**Comment by Danich27rus** [comment]
Intresting, i thoughr i signed CLA 3 weeks ago

**Comment by Danich27rus** [comment]
@Gillibald merge-que failed to merge it

**Pull request merged**